splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -241,24 +247,65 @@ private SdkTracerProvider buildTracerProvider(
 
     //visible for testing
     SpanExporter buildExporter(ConnectionUtil connectionUtil) {
-        String endpoint = config.getBeaconEndpoint() + "?auth=" + config.getRumAccessToken();
         if (!config.isDebugEnabled()) {
             //tell the Zipkin exporter to shut up already. We're on mobile, network stuff happens.
             // we'll do our best to hang on to the spans with the wrapping BufferingExporter.
             ZipkinSpanExporter.baseLogger.setLevel(Level.SEVERE);
             initializationEvents.add(new InitializationEvent("logger setup complete", timingClock.now()));
         }
-        SpanExporter zipkinSpanExporter = getCoreSpanExporter(endpoint);
+
+        if(config.isDiskBufferingDisabled()){
+            SpanExporter spanExporter = buildMemoryBufferingThrottledExporter(connectionUtil);
+            initializationEvents.add(new InitializationEvent("zipkin exporter initialized", timingClock.now()));
+            return spanExporter;
+        }
+        return buildStorageBufferingExporter(connectionUtil);
+    }
+
+    private SpanExporter buildStorageBufferingExporter(ConnectionUtil connectionUtil) {
+        Sender sender = OkHttpSender.newBuilder()
+                .endpoint(getEndpoint())
+                .build();
+        Path filesDir = Paths.get(application.getApplicationContext().getFilesDir().toURI());
+        Path spanFilesPath = filesDir.resolve("spans");
```

Comment:
Can we pass this path to the `getToDiskExporter` method too? Currently it's being computed both here and in `getToDiskExporter`/`ZipkinWriteToDiskExporterFactory`, let's stick to just one place.

Reply:
This changed with the `Path` -> `File` stuff, but it's a good idea and I agree that having it in two spots makes it harder to change later. I'm going to consolidate it.